File: aidlc-table-order/backend/app/dto/menu.py
"""Menu DTOs — request/response models for C4: Menu."""

from pydantic import BaseModel


class CategoryResponse(BaseModel):
    id: str
    name: str
    sort_order: int


class MenuResponse(BaseModel):
    id: str
    name: str
    price: int
    description: str | None = None
    image_url: str | None = None
    sort_order: int
    is_popular: bool
    is_available: bool
    category_id: str


class OptionItemResponse(BaseModel):
    id: str
    name: str
    extra_price: int
    sort_order: int


class OptionGroupResponse(BaseModel):
    id: str
    name: str
    is_required: bool
    items: list[OptionItemResponse]


class MenuDetailResponse(BaseModel):
    id: str
    name: str
    price: int
    description: str | None = None
    image_url: str | None = None
    sort_order: int
    is_popular: bool
    is_available: bool
    category_id: str
    option_groups: list[OptionGroupResponse]


class MenuCreateRequest(BaseModel):
    category_id: str
    name: str
    price: int
    description: str | None = None
    image_url: str | None = None
    is_popular: bool = False
    is_available: bool = True


class MenuUpdateRequest(BaseModel):
    category_id: str | None = None
    name: str | None = None
    price: int | None = None
    description: str | None = None
    image_url: str | None = None
    is_popular: bool | None = None
    is_available: bool | None = None


class MenuReorderRequest(BaseModel):
    menu_ids: list[str]

File: aidlc-table-order/packages/shared/src/types/menu.ts
export interface Category {
  id: string;
  store_id: string;
  name: string;
  sort_order: number;
}

export interface Menu {
  id: string;
  store_id: string;
  category_id: string;
  name: string;
  price: number;
  description: string | null;
  image_url: string | null;
  sort_order: number;
  is_popular: boolean;
  is_available: boolean;
  created_at: string;
  updated_at: string;
}

export interface MenuDetail extends Menu {
  option_groups: OptionGroupWithItems[];
}

export interface OptionGroup {
  id: string;
  store_id: string;
  name: string;
  is_required: boolean;
  created_at: string;
}

export interface OptionItem {
  id: string;
  option_group_id: string;
  name: string;
  extra_price: number;
  sort_order: number;
}

export interface OptionGroupWithItems extends OptionGroup {
  items: OptionItem[];
}

export interface MenusByCategory {
  category: Category;
  menus: Menu[];
}

export interface CreateMenuRequest {
  category_id: string;
  name: string;
  price: number;
  description?: string;
  image_url?: string;
}

export interface UpdateMenuRequest {
  category_id?: string;
  name?: string;
  price?: number;
  description?: string;
  image_url?: string;
  is_popular?: boolean;
  is_available?: boolean;
}

export interface ReorderMenusRequest {
  menu_ids: string[];
}

export interface CreateOptionGroupRequest {
  name: string;
  is_required: boolean;
  items: CreateOptionItemRequest[];
}

export interface CreateOptionItemRequest {
  name: string;
  extra_price: number;
  sort_order?: number;
}

export interface UpdateOptionGroupRequest {
  name?: string;
  is_required?: boolean;
  items?: CreateOptionItemRequest[];
}
